### PR TITLE
Feature/converter/100 add a functionality to currency convereter

### PR DESCRIPTION
**Pull request by LewisLovett**
Currency converted component correctly calculates exchange rate of inputted values and takes the users to the transfer page with the correct props when the transfer button is clicked.

https://trello.com/c/MDyZFpQp/100-add-a-functionality-to-currency-convereter

### CODE REVIEW

File: src/components/ConversionResult/ConversionResult.jsx [hunk 1]
```diff
@@ -15,31 +18,50 @@ import Button from "../Button/Button";
 const ConversionResult = ({
   amount,
   fromCurrency,
-  fromCode,
   toCurrency,
-  toCode,
   rate,
   date,
   handleClick,
+  saveTransferDetails
```
Comment (kirsty-nology) [inline src/components/ConversionResult/ConversionResult.jsx:25]: dont need extra line here

File: src/components/ConversionResult/ConversionResult.jsx [hunk 2]
```diff
@@ -15,31 +18,50 @@ import Button from "../Button/Button";
...
 }) => {
+
```
Comment (kirsty-nology) [inline src/components/ConversionResult/ConversionResult.jsx:27]: add extra line

File: src/components/ConversionResult/ConversionResult.jsx [hunk 3]
```diff
@@ -15,31 +18,50 @@ import Button from "../Button/Button";
...
+  const toAmount = amount * rate;
+  const transferDetails = {fromCurr:fromCurrency,
+    fromAmt:amount,
+    toCurr:toCurrency,
+    toAmt:toAmount,
+    transferRate:rate
+  }
```
Comment (kirsty-nology) [inline src/components/ConversionResult/ConversionResult.jsx:34]: add extra line

File: src/components/ConversionResult/ConversionResult.jsx [hunk 4]
```diff
@@ -15,31 +18,50 @@ import Button from "../Button/Button";
...
+  useEffect(() => {
+    saveTransferDetails.saveTransferDetails(transferDetails);
+  }, []);
+  
+  useEffect(() => {
+    saveTransferDetails.saveTransferDetails(transferDetails);
+  }, [ amount,fromCurrency,toCurrency,rate]);
+
+
```
Comment (kirsty-nology) [inline src/components/ConversionResult/ConversionResult.jsx:43]: remove extra lines above here

File: src/pages/Transfer/Transfer.jsx [hunk 5]
```diff
@@ -24,19 +19,19 @@ const Transfer = ({
       <div className="receipt">
         <h1 className="receipt__head">You Send</h1>
         <div className="receipt__box">
-          <h2 className="receipt__currency">{sendCurr}</h2>
-          <h2 className="receipt__currency">{sendAmount}</h2>
+          <h2 className="receipt__currency">{transferDetails.fromCurr}</h2>
+          <h2 className="receipt__currency">{transferDetails.fromAmt}</h2>
         </div>
```
Comment (kirsty-nology) [inline src/pages/Transfer/Transfer.jsx:24]: `{fromCurr}` instead of `{transferDetails.fromCurr}`

File: src/pages/Transfer/Transfer.jsx [hunk 6]
```diff
@@ -24,19 +19,19 @@ const Transfer = ({
...
         <h1 className="receipt__head">Recipient gets</h1>
```
Comment (kirsty-nology) [inline src/pages/Transfer/Transfer.jsx:25]: `{fromAmt}` instead of `{transferDetails.fromAmt}`

File: src/pages/Transfer/Transfer.jsx [hunk 7]
```diff
@@ -24,19 +19,19 @@ const Transfer = ({
...
         <div className="receipt__box">
-          <h2 className="receipt__currency">{recCurr}</h2>
-          <h2 className="receipt__currency">{recAmount}</h2>
+          <h2 className="receipt__currency">{transferDetails.toCurr}</h2>
+          <h2 className="receipt__currency">{transferDetails.toAmt}</h2>
         </div>
```
Comment (kirsty-nology) [inline src/pages/Transfer/Transfer.jsx:30]: same as above comment

**Review by kirsty-nology — CHANGES_REQUESTED** [review]

File: src/pages/Transfer/Transfer.jsx [hunk 8]
```diff
@@ -1,13 +1,8 @@
...
 const Transfer = ({
-  sendCurr = "GBP - British Pounds",
-  sendAmount = 1000.0,
-  recCurr = "USD - US Dollars",
-  recAmount = 1359.5,
-  rate = 1.3595,
-  fee = 20.0,
+  fee = 0.00,
+  transferDetails
 }) => {
```
[the remark this answers is not in the document]
Comment (LewisLovett) [reply src/pages/Transfer/Transfer.jsx:6]: This change causes the github tests to fail

[the remark this answers is not in the document]
Comment (kirsty-nology) [reply src/pages/Transfer/Transfer.jsx:6]: ok dismiss

**Review by kirsty-nology — APPROVED** [review]

**Review by ChrisWakley — APPROVED** [review]

**Pull request merged**